src/main/java/calculator/Operator.java

```diff
@@ -0,0 +1,32 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum Operator {
+    ADDITION("+", Num::plus),
+    SUBTRACTION("-", Num::minus),
+    MULTIPLICATION("*", Num::times),
+    DIVISION("/", Num::divideBy);
+
+    private final String operator;
+    private final BiFunction<Num, Num, Num> operation;
+
+    Operator(String operator, BiFunction<Num, Num, Num> operation) {
+        this.operator = operator;
+        this.operation = operation;
+    }
+
+    public Num operate(Num first, Num second) {
+        return operation.apply(first, second);
+    }
+
+    public static Operator of(String operator) {
+        return Arrays.stream(values())
+                .filter(object -> object
+                                .operator
+                                .equals(operator))
```

Comment:
```suggestion
                .filter(object -> object.compare(operator))
```
여기서 점 하나만 찍는 규칙 적용해보길 이야기 드린 부분은
제시한 코드와 같은 형태의 코드를 만들어보시길 원했어요. :)